### DISCUSSION TITLE
v0.14.0 Testing checklist

### DISCUSSION THREAD

**Discussion by xoscar** [kubeshop/tracetest#3211] [Release testing]
# Testing Tracetest

This is the checklist on what we should do to assert that Tracetest is working fine on each release version. On each version release, we can copy the contents of this template checklist and open a Github Discussion to start each test.

On Tracetest, we work in two ways to test the entire system and guarantee that everything is working fine. One is through automatic tests, where we run unit, integration, and end-to-end tests on Tracetest CLI, Web UI, and Server.

Another source of tests is the manual tests that we execute on each release, simulating users and checking if everything is ok on these tests.

## Manual Tests

Our manual tests aim to do some [sanity checks](https://en.wikipedia.org/wiki/Sanity_check) to see if the main features are working as expected. Usually, we run it on each Tracetest release.

### Tests to validate on Release Candidate

- Test server installation via CLI
  - [x] Docker Compose and no demo API
  - [x] Docker Compose and demo API
  - [x] Kubernetes and no demo API
  - [x] Kubernetes and demo API
  - [x] Verify that Tracetest is not sending analytics if you have the `TRACETEST_DEV` env for docker install
  - [x] Verify that Tracetest is not sending analytics if you have the `TRACETEST_DEV` env for k8s install

- Test DataStore provisioning and basic test
  - [x] Open WebUI and go to `/settings` page. The provisioned Data Store should be selected
  - [x] Create a test on WebUI that calls a demo API (like [Pokeshop](https://docs.tracetest.io/live-examples/pokeshop/overview) or [Open Telemetry Store](https://docs.tracetest.io/live-examples/opentelemetry-store/overview)). This test should fetch traces correctly and run without errors

- Test Tracetest examples
  - [x] [Jaeger](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-jaeger)
  - [x] [Tempo](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-tempo)
  - [x] [New Relic](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-new-relic)
  - [x] [Lightstep](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-lightstep)
  - [x] [OpenSearch](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-opensearch)
  - [x] [Elastic APM](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-elasticapm-with-elastic-agent)
  - [x] [SignalFX](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-signalfx)
  - [x] [Datadog](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-datadog)
  - [x] [Amazon X-Ray](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-amazon-x-ray)
  - [x] [Honeycomb](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-honeycomb)
  - [x] [Azure Application Insights](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-azure-app-insights)
  - [x] [Signoz](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-signoz)
  - [x] [Dynatrace](https://github.com/kubeshop/tracetest/tree/main/examples/tracetest-dynatrace)

- Test CLI commands (detailed list [here](https://github.com/kubeshop/tracetest/tree/main/testing/cli-e2etest#cli-commands-to-test))

  - [x] Resources: config
  - [x] Resources: datastore
  - [x] Resources: demo
  - [x] Resources: variableset
  - [x] Resources: pollingprofile
  - [x] Resources: analyzer
  - [x] Resources: testrunner
  - [x] Resources: test
  - [x] Resources: testsuite
  - [x] Run tests and testsuites

- Testing benchmark
  - [ ] Run OpenTelemetry Demo tests with the current version and guarantee that the CLI and Server responded correctly

- Test specific features added/changed on this release:
  - [x] validate Settings page https://github.com/kubeshop/tracetest/pull/3194 https://github.com/kubeshop/tracetest/pull/3207
  - [x] update header back button logic https://github.com/kubeshop/tracetest/pull/3199
  - [x] cli version error https://github.com/kubeshop/tracetest/pull/3198
  - [x] empty test and testsuite screens https://github.com/kubeshop/tracetest/pull/3204
  - [x] switching between named and local mode https://github.com/kubeshop/tracetest/pull/3206
  - [x] validate URL in test/testsuite cli summary https://github.com/kubeshop/tracetest/pull/3208
  - [ ] gracefully handle queue disconnection https://github.com/kubeshop/tracetest/pull/3209
  
### Tests to validate on Final Release

- Pipeline
  - [ ] Check if our release pipeline on [Release Tracetest](https://github.com/kubeshop/tracetest/actions/workflows/release-version.yml) workflow on Github Actions worked correctly

- Test CLI updates
  - [ ] MacOS via homebrew
  - [ ] MacOS via curl script
  - [ ] Windows via chocolatey
  - [ ] Windows via manual download

- Docs
  - [ ] Double check [Detailed installation](https://docs.tracetest.io/getting-started/installation) docs and see if everything is documented correctly

- Add Major Changes
  - [ ] Add a summary with major changes to the release page

- Test specific features added/changed on this release:
  - [ ] Add a list of features here

**Comment by jorgeepc** [comment]
### Bugs:
- https://github.com/kubeshop/tracetest/issues/3216
